src/cli/self_update.rs

```diff
@@ -1435,21 +1436,24 @@ pub fn update() -> Result<()> {
         Permit => {}
     }
 
-    let setup_path = prepare_update()?;
-    if let Some(ref p) = setup_path {
-        let version = match get_new_rustup_version(p) {
-            Some(new_version) => parse_new_rustup_version(new_version),
-            None => {
-                err!("failed to get rustup version");
-                process::exit(1);
-            }
-        };
+    match prepare_update()? {
+        Some(ref setup_path) => {
```

Comment:
Does this need to remain `Some(ref setup_path)` now, or could it be `Some(setup_path)` ?

Reply:
To be confirmed. I check and update if needed.